addons/mail/static/src/core/common/attachment_video.js

```diff
@@ -0,0 +1,61 @@
+import { useRef, useState } from "@odoo/owl";
+import { AttachmentUtils } from "@mail/core/common/attachment_utils";
+import { AttachmentActions } from "@mail/core/common/attachment_actions";
+
+/**
+ * @typedef {Object} Props
+ * @property {import("models").Attachment} attachment
+ * @property {number} maxHeight
+ * @property {number} maxWidth
+ * @property {boolean} showDelete
+ * @extends {AttachmentUtils<Props, Env>}
+ */
+export class AttachmentVideo extends AttachmentUtils {
+    static template = "mail.AttachmentVideo";
+    static components = { AttachmentActions };
+    static props = [...AttachmentUtils.props, "attachment", "maxHeight", "maxWidth", "showDelete"];
+
+    setup() {
+        super.setup();
+        this.state = useState({ paused: true });
+        this.videoRefEl = useRef("videoRef");
+        this.seeking = false;
+    }
+
+    onClickPlay(ev) {
+        if (this.videoRefEl.el && this.canPlay) {
+            ev.stopPropagation();
+            this.state.paused = false;
+            this.videoRefEl.el.play();
+            this.videoRefEl.el.setAttribute("controls", "controls");
```

Comment:
```suggestion
            this.videoRefEl.el.controls = true;
```
Couldn't it be used like that or is there a reason why `setAttribute` is used?

Reply:
No particular reason. Just that's how I found it being used in references.